agenta-backend/agenta_backend/models/db_models.py

```diff
@@ -1,303 +1,396 @@
+    variant = relationship("AppVariantDB")
+    revision = Column(Integer)
+    modified_by_id = Column(Integer, ForeignKey("users.id"))
+    modified_by = relationship("UserDB")
+    base_id = Column(Integer, ForeignKey("bases.id"))
+    base = relationship("VariantBaseDB")
+    config_name = Column(String, nullable=False)
+    config_parameters = Column(JSON, nullable=False, default=dict)
+    created_at = Column(
+        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
+    )
+    updated_at = Column(
+        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
+    )
+
+    __tablename__ = "app_variant_revisions"
+
+
+class AppEnvironmentDB(Base):
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    app_id = Column(Integer, ForeignKey("app_db.id"))
+    app = relationship("AppDB")
+    name = Column(String)
+    user_id = Column(Integer, ForeignKey("users.id"))
+    user = relationship("UserDB")
+    revision = Column(Integer)
+    deployed_app_variant_id = Column(Integer)  # reference to app_variant
+    deployed_app_variant_revision_id = Column(
+        Integer, ForeignKey("app_variant_revisions.id")
+    )
+    deployed_app_variant_revision = relationship("AppVariantRevisionsDB")
+    deployment_id = Column(Integer)  # reference to deployment
+    created_at = Column(
+        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
+    )
+
+    __tablename__ = "environments"
+
+
+class AppEnvironmentRevisionDB(Base):
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    environment_id = Column(Integer, ForeignKey("environments.id"))
+    environment = relationship("AppEnvironmentDB")
+    revision = Column(Integer)
+    modified_by_id = Column(Integer, ForeignKey("users.id"))
+    modified_by = relationship("UserDB")
+    deployed_app_variant_revision_id = Column(
+        Integer
+    )  # reference to app_variant_revision
+    deployment_id = Column(Integer)  # reference to deployment
+    created_at = Column(
+        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
+    )
+
+    __tablename__ = "environments_revisions"
+
+
+class TemplateDB(Base):
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    type = Column(String, default="image")
+    template_uri = Column(String)
+    tag_id = Column(Integer)
+    name = Column(String, unique=True)
+    repo_name = Column(String)
+    title = Column(String)
+    description = Column(String)
+    size = Column(Integer)
+    digest = Column(String)  # sha256 hash of image digest
+    last_pushed = Column(
+        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
+    )
+
+    __tablename__ = "templates"
+
+
+class TestSetDB(Base):
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    name = Column(String)
+    app_id = Column(Integer, ForeignKey("app_db.id"))
+    app = relationship("AppDB")
+    csvdata = Column(JSON)  # List of dictionaries
```

Comment:
I don't think this is the best solution. We have already seen that some users have very large test sets. Having these in one json in postgres would come with lots of issues. I think a better solution is to have this on a different table, no?

Reply:
No as previously communicated, the best solution is to migrate to datapoints and a separate table. Not sure if we need to do it now though.